### ISSUE TITLE
Makind DO loop syntax consitent with the DO CONCURRENT one

### ISSUE THREAD

**Issue by PierUgit** [j3-fortran/fortran_proposals#338]
I'm not sure this proposal doesn't already exist, but I can't find it. If it exists, feel free to delete this issue.

It would good to allow for the simple do loop the same syntax as the do concurrent loop, not only for internal consistency of the language, but mainly to allow the on-the-fly declaration of the index:
```fortran
do (integer :: i = 1:n)
   ...
end do
```
For an even better consistency, the `sequential` keyword could be used:
```
do sequential (integer :: i = 1:n)
   ...
end do
```

**Comment by certik** [comment]
Yes, I know I've seen this exact proposal before, but can't find it right now so we'll leave this open. Yes, going from serial `do` to concurrent `do` requires changing the syntax, which is very annoying.

**Comment by han190** [comment]
This proposal could be very useful to long nested loops. Julia has something like
```Julia
for i=1:n, j=1:m
end
```
and in Fortran we could have something similar
```Fortran
do sequential (i=1:n, j=1:m)
end do
```
More aggressively we could adopt the Fortran 2023 array subscript feature,
```Fortran
do sequential (integer :: i,j,k = @[1,1,1]:[n,m,l])
end do
```
